app/src/main/kotlin/com/wire/android/util/UriUtil.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.util
+
+import java.net.URLDecoder
+
+fun containsSchema(url: String): Boolean {
+    val regexPattern = Regex(".+://.+")
+
+    return regexPattern.matches(url)
+}
+
+fun normalizeLink(url: String): String {
+    val normalizedUrl = URLDecoder.decode(url, "UTF-8") // Decode URL to human-readable format
+
+    return if (containsSchema(normalizedUrl)) {
+        normalizedUrl
+    } else {
+        "https://$normalizedUrl"
```

Comment:
question: if the link already have something like, "/google.com" then it will turn into "https:///google.com" and it will fail to open

Reply:
/google.com is not a URL

Reply:
but what if we send `[google](/google.com)` then it will go `https:///google.com` as @MohamadJaara mentioned, no? 🤔

Reply:
I'd blame whoever pasted the invalid URI there to be honest 😄.

Would be a nice to have for the rare cases where people mistype that hard, but not sure if worth the effort